devicetypes/fibargroup/fibaro-single-switch-2-zw5.src/fibaro-single-switch-2-zw5.groovy

```diff
@@ -0,0 +1,513 @@
+    def failed = []
+    def incorrect = []
+    def notSynced = []
+    parameterMap().each {
+        if (state."$it.key"?.state == "incorrect" ) {
+            incorrect << it
+        } else if ( state."$it.key"?.state == "failed" ) {
+            failed << it
+        } else if ( state."$it.key"?.state in ["inProgress","notSynced"] ) {
+            notSynced << it
+        }
+    }
+
+    if (failed) {
+        multiStatusEvent("Sync failed! Verify parameter: ${failed[0].num}", true, true)
+    } else if (incorrect) {
+        multiStatusEvent("Sync mismatch! Verify parameter: ${incorrect[0].num}", true, true)
+    } else if (notSynced) {
+        multiStatusEvent("Sync incomplete! Open settings and tap Done to try again.", true, true)
+    } else {
+        if (device.currentValue("multiStatus")?.contains("Sync")) { multiStatusEvent("Sync OK.", true, true) }
+    }
+}
+
+private multiStatusEvent(String statusValue, boolean force = false, boolean display = false) {
+    if (!device.currentValue("multiStatus")?.contains("Sync") || device.currentValue("multiStatus") == "Sync OK." || force) {
+        sendEvent(name: "multiStatus", value: statusValue, descriptionText: statusValue, displayed: display)
+    }
+}
+
+//event handlers related to configuration and sync
+def zwaveEvent(physicalgraph.zwave.commands.configurationv2.ConfigurationReport cmd) {
+    def paramKey = parameterMap().find( {it.num == cmd.parameterNumber } ).key
+    logging("Parameter ${paramKey} value is ${cmd.scaledConfigurationValue} expected " + state."$paramKey".value, "info")
+    state."$paramKey".state = (state."$paramKey".value == cmd.scaledConfigurationValue) ? "synced" : "incorrect"
+    syncNext()
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.applicationstatusv1.ApplicationRejectedRequest cmd) {
+    logging("rejected request!","warn")
+    for ( param in parameterMap() ) {
+        if ( state."$param.key"?.state == "inProgress" ) {
+            state."$param.key"?.state = "failed"
+            break
+        }
+    }
+}
+
+//event handlers
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+    //ignore
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+    logging("SwitchBinaryReport received, value: ${cmd.value} ","info")
+    sendEvent([name: "switch", value: (cmd.value == 0 ) ? "off": "on"])
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.meterv3.MeterReport cmd) {
+    logging("MeterReport received, value: ${cmd.scaledMeterValue} scale: ${cmd.scale} ","info")
+    switch (cmd.scale) {
+        case 0: sendEvent([name: "energy", value: cmd.scaledMeterValue, unit: "kWh"]); break;
+        case 2: sendEvent([name: "power", value: cmd.scaledMeterValue, unit: "W"]); break;
+    }
+    multiStatusEvent("${(device.currentValue("power") ?: "0.0")} W | ${(device.currentValue("energy") ?: "0.00")} kWh")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.centralscenev1.CentralSceneNotification cmd) {

```

Comment:
Again, does the device actually support these things? These buttons don't seem to be exposed in any way and the device doesn't seem to use any multichannel capabilities.